packages/block-library/src/template-part/edit/selection-modal.js

```diff
@@ -76,40 +89,53 @@ export default function TemplatePartSelectionModal( {
 	);
 
 	return (
-		<>
-			<div className="block-library-template-part__selection-content">
-				{ !! templartPartsAsBlockPatterns.length && (
-					<div>
-						<h2>{ __( 'Existing template parts' ) }</h2>
-						<BlockPatternsList
-							blockPatterns={ templartPartsAsBlockPatterns }
-							shownPatterns={ shownTemplateParts }
-							onClickPattern={ ( pattern ) => {
-								onTemplatePartSelect( pattern.templatePart );
-							} }
-						/>
-					</div>
-				) }
+		<div className="block-library-template-part__selection-content">
+			<div className="block-library-template-part__selection-search">
+				<SearchControl
+					onChange={ setSearchValue }
+					value={ searchValue }
+					label={ __( 'Search for replacements' ) }
+					placeholder={ __( 'Search' ) }
+				/>
+			</div>
+			{ !! filteredTemplateParts.length && (
+				<div>
+					<h2>{ __( 'Existing template parts' ) }</h2>
+					<BlockPatternsList
+						blockPatterns={ filteredTemplateParts }
+						shownPatterns={ shownTemplateParts }
+						onClickPattern={ ( pattern ) => {
+							onTemplatePartSelect( pattern.templatePart );
+						} }
+					/>
+				</div>
+			) }
 
-				{ !! blockPatterns.length && (
-					<div>
-						<h2>{ __( 'Patterns' ) }</h2>
-						<BlockPatternsList
-							blockPatterns={ blockPatterns }
-							shownPatterns={ shownBlockPatterns }
-							onClickPattern={ ( pattern, blocks ) => {
-								if ( isReplacingTemplatePartContent ) {
-									replaceInnerBlocks( clientId, blocks );
-								} else {
-									createFromBlocks( blocks, pattern.title );
-								}
+			{ !! filteredBlockPatterns.length && (
+				<div>
+					<h2>{ __( 'Patterns' ) }</h2>
+					<BlockPatternsList
+						blockPatterns={ filteredBlockPatterns }
+						shownPatterns={ shownBlockPatterns }
+						onClickPattern={ ( pattern, blocks ) => {
+							if ( isReplacingTemplatePartContent ) {
+								replaceInnerBlocks( clientId, blocks );
+							} else {
+								createFromBlocks( blocks, pattern.title );
+							}
 
-								onClose();
-							} }
-						/>
-					</div>
+							onClose();
+						} }
+					/>
+				</div>
+			) }
+
+			{ ! filteredTemplateParts.length &&
+				! filteredBlockPatterns.length && (
+					<HStack alignment="center">
```

Comment:
I guess we should use `VStack` here, no?

Reply:
The "No results found" paragraph is centered horizontally, so I thought `HStack` was the right choice. Any reason for using `VStack` instead?

Reply:
Not a strong opinion, but for me it seemed a single child of a vertical stack, which would need to be text aligned in the center. What will be shown if for example we added a border to that now and in the difference from VStack.

Reply:
I decided to leave `HStack` for now.